### DISCUSSION TITLE
"Failed to update edge function: Function body is invalid" When Updating Edge Function

### DISCUSSION THREAD

**Discussion by jameswilkinson1** [supabase/supabase#21611] [Questions]
Hi all, 

Anybody else seen "Failed to update edge function: Function body is invalid" when attempting to remove the JWT authorization from an Edge Function?

I was deploying a lot of Edge Functions yesterday and didn't have this issue. I deleted all my Edge Functions and redeployed them and I still have the same issue. 

Anybody got any insight? It'd be greatly appreciated.

Thanks all!

**Comment by nicolonic** [comment]
Experiencing same issue right now.

**Reply by jameswilkinson1** [reply]
Thanks mate. I've emailed Supabase support but they state it will be 24-48 hours before I get a reply. I guess it's on the Supabase side then if it's the same for both of us.

**Comment by nicolonic** [comment]
I did the same

*Nic Nicolo*
***@***.***
Linkedin <https://www.linkedin.com/in/nicholas-nicolo/>

**Comment by jameswilkinson1** [comment]
UPDATE: Supabase support narrowed down the issue to a Change in Supabase's Management API.

Supabase have told me they are working on the fix now. I will post again when they give me another updated (presumably the issue has been resolved).

**Reply by jareddr** [reply]
Thanks! Having the same issue, have been browsing the web for an hour looking for a solution.

**Reply by jareddr** [reply]
Since I have a group of people trying to use a similar feature here, I wonder if anyone could help me answer this question.  

I'm trying to disable the "Enforce JWT verification" option on my edge function because the function is supposed to catch a webhook from apple pay which won't have an Authorization header.  Currently, when I visit my endpoint in the browser I get a 'Missing authorization header' response.  Is this the intended use of disabling the "Enforce JWT verification" setting?

Thanks and sorry for the tangent.

**Comment by nicolonic** [comment]
The support team got back to me today. Apparently it has to do with a
release they made. Waiting for them to push an update still.

On Wed, Feb 28, 2024 at 3:33 PM Jared ***@***.***> wrote:

> Since I have a group of people trying to use a similar feature here, I
> wonder if anyone could help me answer this question.
>
> I'm trying to disable the "Enforce JWT verification" option on my edge
> function because the function is supposed to catch a webhook from apple pay
> which won't have an Authorization header. Currently, when I visit my
> endpoint in the browser I get a 'Missing authorization header' response. Is
> this the intended use of disabling the "Enforce JWT verification" setting?
>
> Thanks and sorry for the tangent.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/supabase/discussions/21611#discussioncomment-8622962>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMAMXAUUVLFTCFWJI44UMGDYV6IAZAVCNFSM6AAAAABD4LREYKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMMRSHE3DE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Nic Nicolo*
***@***.***
(423) 435-8764
Linkedin <https://www.linkedin.com/in/nicholas-nicolo/>
